ui/src/features/project/pipelines/freight/use-soak-time-counter.ts

```diff
@@ -0,0 +1,69 @@
+import { Duration } from 'date-fns';
+import { useEffect, useMemo, useState } from 'react';
+
+import { queryCache } from '@ui/features/utils/cache';
+
+import { durationToSeconds } from './duration-to-seconds';
+
+export const useSoakTimeCounter = (soakTime?: Duration) => {
+  const [counter, setCounter] = useState(soakTime);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setCounter((prev) => {
+        if (!prev) {
+          return prev;
+        }
+
+        const newCounter = { ...prev };
+        if (newCounter.seconds !== undefined) {
+          newCounter.seconds -= 1;
+        }
+
+        if ((newCounter?.seconds || 0) < 0) {
+          newCounter.seconds = 59;
+
+          if (newCounter.minutes !== undefined) {
+            newCounter.minutes -= 1;
+          }
+        }
+
+        if ((newCounter?.minutes || 0) < 0) {
+          newCounter.minutes = 59;
+
+          if (newCounter.hours !== undefined) {
+            newCounter.hours -= 1;
+          }
+        }
```

Comment:
We don't need to update every digit here, just recalculate time left using https://date-fns.org/v4.1.0/docs/differenceInMilliseconds

Reply:
I don't understand how that would help here. I think with that, at some point we still need to do some transformations to have formatted date

Reply:
We want to include this PR in 1.5, so let's discuss it later.